toto/verifylib.py

```diff
@@ -303,261 +315,304 @@ def verify_match_rule(rule, source_type, source_link, target_links):
-    or if the hashes are equal (were not modified).
+    RuleVerficationFailed if path pattern matches files in artifact queue.
 
   <Side Effects>
+    Uses fnmatch.filter which translates a glob pattern to re.
+
+  <Returns>
     None.
+    In contrast to other rule types, the DELETE rule does not
+    remove matched files from the artifact queue, because it MUST not match
+    files in order to pass.
 
   """
   # FIXME: Validate rule format
+  path_pattern = rule[1]
+  matched_artifacts = fnmatch.filter(artifact_queue, path_pattern)
+  if matched_artifacts:
+    raise RuleVerficationFailed("Rule {0} failed, artifacts {1} "
+        "were not deleted".format(rule, matched_artifacts))
+
+
+# def verify_modify_rule(rule, link):
+#   """
+#   <Purpose>
+#     Verifies that the path (2nd element in rule list) is found in the materials
+#     and products list of the passed Link object and that the hashes of the
+#     according material and product are not equal, i.e. the file was modified in
+#     the step the rule was defined for.
+
+#   <Arguments>
+#     rule:
+#             The rule to be verified. Format is: ["MODIFY", "<path>"]
+
+#     link:
+#             The Link object for the Item (Step or Inspection) that contains
+#             the rule.
+
+#   <Exceptions>
+#     raises an Exception if the rule does not conform with the rule format.
+#     raises an if a matchrule does not verify.
+#     TBA (see https://github.com/in-toto/in-toto/issues/6)
+
+#     RuleVerficationFailed if path is not found in the materials or products
+#     or if the hashes are equal (were not modified).
 
-  path = rule[1]
+#   <Side Effects>
+#     None.
 
-  if (path not in link.materials.keys()):
-    raise RuleVerficationFailed("'%s' "
-        "not found in materials of link '%s'" % (path, link.name))
+#   """
+#   # FIXME: Validate rule format
 
-  if (path not in item_link.products.keys()):
-    raise RuleVerficationFailed("'%s' "
-        "not found in products of link '%s'" % (path, link.name))
+#   path = rule[1]
 
-  if (ComparableHashDict(link.materials[path]) == \
-      ComparableHashDict(link.products[path])):
-    raise RuleVerficationFailed("hashes of product and material '%s' of link "
-        "'%s' match - should have been modified" % (path, link.name))
+#   if (path not in link.materials.keys()):
+#     raise RuleVerficationFailed("'%s' "
+#         "not found in materials of link '%s'" % (path, link.name))
 
+#   if (path not in item_link.products.keys()):
+#     raise RuleVerficationFailed("'%s' "
+#         "not found in products of link '%s'" % (path, link.name))
 
-def _verify_rules(rules, source_type, source_link, target_links):
+#   if (ComparableHashDict(link.materials[path]) == \
+#       ComparableHashDict(link.products[path])):
+#     raise RuleVerficationFailed("hashes of product and material '%s' of link "
+#         "'%s' match - should have been modified" % (path, link.name))
```

Comment:
Could we take this code out of this commit? so we can have a cleaner diff when this function is added.